sources/packages/backend/libs/services/src/students-assessments/assessment-sequential-processing.service.ts

```diff
@@ -343,4 +385,139 @@ export class AssessmentSequentialProcessingService {
+      .addSelect("SUM(sfasApplication.sbsd_award)", "SBSD")
+      .addSelect("SUM(sfasApplication.csgd_award)", "CSGD")
+      .addSelect("SUM(sfasApplication.bcag_award)", "BCAG")
+      .innerJoin("sfasApplication.individual", "sfasStudent")
+      .where("lower(sfasStudent.lastName) = lower(:lastName)", { lastName })
+      .andWhere("sfasStudent.sin = :sin", { sin })
+      .andWhere("sfasStudent.birthDate = :birthDate", { birthDate })
+      .andWhere("sfasApplication.startDate >= :startDate", {
+        startDate: programYearStartDate,
+      })
+      .andWhere("sfasApplication.endDate < :endDate", {
+        endDate: firstAssessmentDate,
+      });
+
+    const awards =
+      await sfasApplicationAwards.getRawOne<SFASApplicationAwards>();
+    const totals: AwardTotal[] = [];
+    if (awards) {
+      Object.entries(awards).forEach(([key, value]) => {
+        if (+value > 0) {
+          totals.push({
+            offeringIntensity: OfferingIntensity.fullTime,
+            valueCode: key,
+            total: +value,
+          });
+        }
+      });
+    }
+    return totals;
+  }
+
+  /**
+   * Get SFAS part time (SAIL) application awards totals.
+   * @param assessment Current assessment being processed.
+   * @param firstAssessmentDate Date of the first assessment date of the current application.
+   * @returns SFAS application part time awards totals.
+   */
+  async getProgramYearSAILAwardsTotals(
+    assessment: StudentAssessment,
+    firstAssessmentDate: string,
+  ): Promise<AwardTotal[]> {
+    const lastName = assessment.application.student.user.lastName;
+    const sin = assessment.application.student.sinValidation.sin;
+    const birthDate = assessment.application.student.birthDate;
+    const programYearStartDate = assessment.application.programYear.startDate;
+    const sailApplicationAwards = this.sfasPartTimeApplicationsRepo
+      .createQueryBuilder("sfasPTApplication")
+      .select("SUM(sfasPTApplication.csgp_award)", "CSGP")
+      .addSelect("SUM(sfasPTApplication.sbsd_award)", "SBSD")
+      .addSelect("SUM(sfasPTApplication.csgd_award)", "CSGD")
+      .addSelect("SUM(sfasPTApplication.bcag_award)", "BCAG")
+      .addSelect("SUM(sfasPTApplication.cspt_award)", "CSPT")
+      .innerJoin("sfasPTApplication.individual", "sfasStudent")
+      .where("lower(sfasStudent.lastName) = lower(:lastName)", { lastName })
+      .andWhere("sfasStudent.sin = :sin", { sin })
+      .andWhere("sfasStudent.birthDate = :birthDate", { birthDate })
+      .andWhere("sfasPTApplication.startDate >= :startDate", {
+        startDate: programYearStartDate,
+      })
+      .andWhere("sfasPTApplication.endDate < :endDate", {
+        endDate: firstAssessmentDate,
+      });
+    const awards =
+      await sailApplicationAwards.getRawOne<SAILApplicationAwards>();
+    const totals: AwardTotal[] = [];
+    if (awards) {
+      Object.entries(awards).forEach(([key, value]) => {
+        if (+value > 0) {
```

Comment:
value can be technically null as it is a nullable field in DB.

Please check how the typeorm returns if an award is null and add check based on that.

![image](https://github.com/bcgov/SIMS/assets/54600590/9f3fc6ab-8a8b-485f-90b2-0c5c61cbf21d)